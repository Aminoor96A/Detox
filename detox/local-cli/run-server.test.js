describe('run-server', () => {
  let DetoxServer;
  let command;

  beforeEach(() => {
    jest.mock('../src/server/DetoxServer');
    DetoxServer = require('../src/server/DetoxServer');
    command = require('./__helpers__/CommandBuilder').init().withCommand('run-server');
  });

  it('starts the server', async () => {
    await command.exec();
    expect(DetoxServer).toHaveBeenCalledWith(expect.objectContaining({ port: 8099 }));
  });

  it('throws if the port number is out of range', async () => {
    jest.spyOn(process, 'exit'); // otherwise tests are aborted

    expect(command.withEnv({ DETOX_PORT: 99999 }).exec()).rejects.toThrowErrorMatchingSnapshot();
    expect(command.withArgs('-p 100000').exec()).rejects.toThrowErrorMatchingSnapshot();
    expect(command.withArgs('-p 0').exec()).rejects.toThrowErrorMatchingSnapshot();
    expect(command.withArgs('-p DEADBEEF').exec()).rejects.toThrowErrorMatchingSnapshot();
    expect(DetoxServer).not.toHaveBeenCalled();
  });
});
