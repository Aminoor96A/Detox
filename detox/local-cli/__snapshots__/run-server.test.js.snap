// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`run-server throws if the port number is out of range 1`] = `"The port should be between 1 and 65535, got 99999"`;

exports[`run-server throws if the port number is out of range 2`] = `"The port should be between 1 and 65535, got 100000"`;

exports[`run-server throws if the port number is out of range 3`] = `"The port should be between 1 and 65535, got 0"`;

exports[`run-server throws if the port number is out of range 4`] = `"The port should be between 1 and 65535, got NaN"`;
